src/validation.cpp

```diff
@@ -930,6 +945,18 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
     if(!fDryRun)
         GetMainSignals().TransactionAddedToMempool(ptx);
 
+    boost::posix_time::ptime finish = boost::posix_time::microsec_clock::local_time();
+    boost::posix_time::time_duration diff = finish - start;
+    statsClient.timing("AcceptToMemoryPool_ms", diff.total_milliseconds(), 1.0f);
+    statsClient.gauge("transactions.txInMemoryPool", pool.size(), 0.1f);
```

Comment:
This doens't look like it belongs here. It only updates this metric when a TX comes in but does not update it when blocks remove the TXs from mempool. Better to do these mempool related metrics periodically (same background thread as for the utxo stats).